routes/doctor.js

```diff
@@ -0,0 +1,95 @@
+const getReply = function(chatResponse, done) {
+    let result = chatResponse.result;
+    const parameters = result.parameters;
+    const intentName = result.metadata.intentName;
+    if (intentName === 'CheckMalariaSymptoms') {
+        CheckMalariaSymptoms(parameters, function(reply) {
+            done(reply);
+        });
+    } else if (intentName === 'medicineSafeCheckQuestion') {
+        const contextParameters = result.contexts[0].parameters;
+        medicineSafeCheck(contextParameters, function(reply) {
+            done(reply);
+        });
+    } else if (intentName === 'medicineSideEffects') {
+        medicineSideEffects(parameters, function(reply) {
+            done(reply);
+        });
+    }
+}
+
+
+const CheckMalariaSymptoms = function(parameters, done) {
+
+    const symptomScores = {
+        chills: 2,
+        fatigue: 2,
+        fever: 2,
+        malaise: 2,
+        shivering: 2,
+        sweating: 2,
+        headache: 1,
+        nausea: 1,
+        'pain in muscles': 1,
+        'faster heart rate': 1
+    }
+
+    const symptoms = parameters.Symptoms;
+    let length = symptoms.length;
+    let malariaRisk = 0;
+
+    symptoms.forEach(function(symptom) {
+        malariaRisk += symptomScores[symptom];
+        if (--length === 0) {
+            generateSymptomsReply(malariaRisk, function(reply) {
+                done(reply);
+            });
+        }
+    });
+}
+
+const generateSymptomsReply = function(malariaRisk, done) {
+    let reply;
+
+    if (malariaRisk > 6) {
+        reply = 'You seem to have quite a lot of symptoms related to malaria. Malaria can be a threat to life, please visit your doctor Immediately.';
+    } else if (malariaRisk > 3) {
+        reply = 'There are chances of you being infected with malaria. Malaria can be a threat to life, It is always recommended you visit your doctor if you have any doubts.';
+    } else {
+        reply = 'It is less likely that you are infected with malaria. However it is always recommended you visit a doctor since malaria can turn out to be life threatening.';
+    }
+    done(reply);
+}
+
+const medicineSafeCheck = function(contextParameters, done) {
+    const gender = contextParameters.gender;
+    const medicine = contextParameters.medicine;
+    const age = contextParameters.age.amount;
+    /**
+     * Feature incomplete
```

Comment:
No, this is for future, not in GSoC. For this we need some verified information about medication based on age and gender,  a decision tree based on age and gender has to be added. I did not find any such reliable information so added a generic reply.